src/utils/wallet.js

```diff
@@ -512,20 +934,37 @@ class Wallet {
     }
 
     async deleteRecoveryMethod({ kind, publicKey }) {
+        const res = await this.removeAccessKey(publicKey)
+        console.log('deleteRecoveryMethod', res)
         await this.postSignedJson('/account/deleteRecoveryMethod', {
             accountId: this.accountId,
             kind,
             publicKey
         })
-        await this.removeAccessKey(publicKey)
     }
 
-    async recoverAccountSeedPhrase(seedPhrase) {
+    async recoverAccountSeedPhrase(seedPhrase, use2fa = false, accountId, fromSeedPhraseRecovery = false) {
         const { publicKey, secretKey } = parseSeedPhrase(seedPhrase)
+
+        console.log('recovering account with publicKey', publicKey)
+
         const accountIds = await getAccountIds(publicKey)
+        // if we don't find accountIds we can push one from the link (email/sms ONLY)
+        if (accountId && !accountIds.includes(accountId)) {
+            accountIds.push(accountId)
+        }
 
         if (!accountIds.length) {
-            throw new WalletError('Cannot find matching public key', 'account.recoverAccount.errorInvalidSeedPhrase', { publicKey })
+            // hopefully an unlikely scenario after we fix indexer
+            const userProvidedAccountId = window.prompt('Cannot find Account Name. Please enter the name of the account to recover.')
```

Comment:
let's not use `window.prompt` for anything

Reply:
Any thoughts on how we should capture accountId until the indexer is fixed?